### PR TITLE
Enable properties which supported in sdk

### PR DESCRIPTION
**Pull request by zhichengliu12581**
enable properties which supported in sdk
related [pr](https://github.com/Azure/azure-sdk-for-java/pull/22604)

### CODE REVIEW

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          binder: servicebus-2
+          destination: queue2
+        queueSupply-out-0:
+          binder: servicebus-2
+          destination: queue2
+      binders:
+        servicebus-1:
+          type: servicebus-topic
+          default-candidate: true
+          environment:
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: is this added new?
very wired

Comment (zhichengliu12581) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: Yes, in order to add an integration test for the configuration of the added `serviceBusReceiveMode` parameter, I can delete it if it is not necessary😅

**Comment by stliu** [comment]
docs?

**Comment by zhichengliu12581** [comment]
@stliu I'm waiting for the review for this [pr](https://github.com/Azure/azure-sdk-for-java/pull/22778), which have conflict with this pr. I plan to modify the document and changelog after that pr merge. Or I can add it temporarily, and then modify it again after the pr merge?

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml (hunk 1 above)
Comment (stliu) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: never mind, I was wrong, just asked Yi Liu about this

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusProperties.java [hunk 2]
```diff
@@ -16,6 +18,8 @@
 
     private String connectionString;
 
+    private AmqpRetryOptions retryOptions = new AmqpRetryOptions().setTryTimeout(ServiceBusConstants.OPERATION_TIMEOUT);
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusProperties.java:21]: Is this default value for servicebus sdk client? If so, seems we don't set it here

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java [hunk 3]
```diff
@@ -72,6 +72,8 @@ public ServiceBusTopicClientFactory topicClientFactory(
         Assert.notNull(connectionString, "Service Bus connection string must not be null");
 
         DefaultServiceBusTopicClientFactory clientFactory = new DefaultServiceBusTopicClientFactory(connectionString, properties.getTransportType());
+        clientFactory.retryOptions(properties.getRetryOptions());
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java:75]: how about using setRetryOptions to keep all function names consistent?

File: sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusAutoConfigurationTest.java [hunk 4]
```diff
@@ -97,6 +98,23 @@ public void testTransportTypeWithAmqpWebSockets() {
             });
     }
 
+    @Test
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusAutoConfigurationTest.java:101]: Can we add UTs in the queue&topic autoconfigurtion to see if the client factory beans are created as the way the properties are configured?

File: sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md [hunk 5]
```diff
@@ -49,6 +49,7 @@ spring.cloud.azure.resource-group | Name of Azure resource group | Yes |
 spring.cloud.azure.region | Region name of the Azure resource group, e.g. westus | Yes | 
 spring.cloud.azure.servicebus.namespace | Service Bus Namespace. Auto creating if missing | Yes |
 spring.cloud.azure.servicebus.transportType | Service Bus transportType, supported value of `AMQP` and `AMQP_WEB_SOCKETS` | No | `AMQP`
+spring.cloud.azure.servicebus.retry-Options | Service Bus retry options | No |
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:52]: Seems we have a default value?

File: sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md [hunk 6]
```diff
@@ -106,6 +107,16 @@ Set it to `true` if a message that trigger any exception in consumer will be for
...
+
+**_receiveMode_**
+
+The modes for receiving messages.
+
+`PEEK_LOCK`, received message is not deleted from the queue or subscription, instead it is temporarily locked to the receiver, making it invisible to other receivers.
+
+`RECEIVE_AND_DELETE`, received message is removed from the queue or subscription and immediately deleted.
+
+Default: `PEEK_LOCK`
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:119]: Should we add introdution for maxConcurrentCalls as well?

Comment (zhichengliu12581) [reply sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:119]: It already has an introduction to maxConcurrentCalls [here](https://github.com/zhichengliu12581/azure-sdk-for-java/blob/dc8756197ea3c09fec4caa9215933105bc5eebab/sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md#L94)

File: sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md [hunk 7]
```diff
@@ -4,7 +4,15 @@
...
+- Support configuration of `retryOptions` for ServiceBusClientBuilder with property of `spring.cloud.azure.servicebus.retry-options`, these parameters can be modified: `maxRetries`, `delay`, `maxDelay`, `tryTimeout`, `Mode`.
+- Support configuration of `maxConcurrentCalls` for ServiceBusClientConfig with property of `spring.cloud.stream.servicebus.<queue or topic>.bindings.<channelName>.consumer.maxConcurrentCalls`.
+- Support configuration of `maxConcurrentSessions` for ServiceBusClientConfig with property of `spring.cloud.stream.servicebus.<queue or topic>.bindings.<channelName>.consumer.maxConcurrentSessions`.
+- Support configuration of `serviceBusReceiveMode` for ServiceBusClientConfig with property of `spring.cloud.stream.servicebus.<queue or topic>.bindings.<channelName>.consumer.serviceBusReceiveMode`, supported values are `PEEK_LOCK` and `RECEIVE_AND_DELETE`.
+- Support configuration of `enableAutoComplete` for ServiceBusClientConfig with property of `spring.cloud.stream.servicebus.queue.bindings.<channelName>.consumer.enableAutoComplete`.
+### Breaking Changes
+- Deprecated the `spring.cloud.stream.servicebus.queue.bindings.<channelName>.consumer.concurrency` property,
```
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md:14]: Deprecated --> Deprecate

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java [hunk 8]
```diff
@@ -63,6 +63,8 @@ public ServiceBusQueueClientFactory queueClientFactory(
         Assert.notNull(connectionString, "Service Bus connection string must not be null");
 
         DefaultServiceBusQueueClientFactory clientFactory = new DefaultServiceBusQueueClientFactory(connectionString, properties.getTransportType());
+        clientFactory.retryOptions(properties.getRetryOptions());
+
```
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java:67]: can we delete the empty line here?

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java [hunk 9]
```diff
@@ -72,6 +72,8 @@ public ServiceBusTopicClientFactory topicClientFactory(
...
+
```
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java:76]: can we delete the empty line here?

**Review by yiliuTo — APPROVED** [review]
lgtm

**Pull request merged**